pkg/cfn/builder/cluster.go

```diff
@@ -58,7 +58,7 @@ func (c *ClusterResourceSet) AddAllResources() error {
 	c.vpcResourceSet.AddOutputs()
 	clusterSG := c.addResourcesForSecurityGroups(vpcResource)
 
-	if privateCluster := c.spec.PrivateCluster; privateCluster.Enabled {
+	if privateCluster := c.spec.PrivateCluster; privateCluster.Enabled && !privateCluster.SkipEndpointCreation {
```

Comment:
same here 🙈

Reply:
again, this is idiomatic go code structure. :) It's a variable only relevant to this if statement.